### PR TITLE
fix: corrected tracepoints descriptions

### PR DESCRIPTION
**Pull request by xygyo77**
## Description

Fixed description of trace points that deviated from implementation.

## Related links

[https://tier4.atlassian.net/browse/RT2-1578](https://tier4.atlassian.net/browse/RT2-1578)

## Pre-review checklist for the PR author

- [x] I've confirmed the [contribution guidelines](https://github.com/tier4/caret/blob/main/.github/CONTRIBUTING.md).
- [x] The PR is ready for review.

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR has been reviewed.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.
- [x] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

### CODE REVIEW

File: docs/recording/trace_filtering.md [hunk 1]
```diff
@@ -58,3 +58,8 @@ export CARET_IGNORE_TOPICS=\
 # /parameter_events:\
 # "
 ```
+
+<prettier-ignore-start>
+!!! Info
+        If the node cannot be identified from the trace point information, the trace is collected as not applicable to the filter.
```
Comment (ymski) [inline docs/recording/trace_filtering.md:64]: I made an alternative. This wording may be easier for users to understand.

```suggestion
        If the node cannot be identified from the trace point information, the trace will be collected without applying the filter.
```

File: docs/design/event_and_latency_definitions/publisher.md [hunk 2]
```diff
@@ -77,3 +77,8 @@ See also
 - [Trace points | dds_write](../trace_points/runtime_trace_points.md#ros2_caretdds_write)
 - [Trace points | bind_addr_to_addr](../trace_points/runtime_trace_points.md#ros2_caretdds_bind_addr_to_addr)
 - [Trace points | bind_addr_to_stamp](../trace_points/runtime_trace_points.md#ros2_caretdds_bind_addr_to_stamp)
+
+<prettier-ignore-start>
+!!! Info
+        If it is not after jazzy, it will not appear in generic_publish().
```
Comment (ymski) [inline docs/design/event_and_latency_definitions/publisher.md:83]: In humble, only the trace points of the rclcpp layer are available for generic_publish().

File: docs/design/trace_points/runtime_trace_points.md
Comment (ymski) [inline docs/design/trace_points/runtime_trace_points.md:1]: rclcpp_intra_publish is classified as Extended tracepoints in humble, but is classified as Built-in tracepoints in iron and later. Please reflect this in the documentation as well.

File: docs/design/trace_points/initialization_trace_points.md [hunk 3]
```diff
@@ -98,6 +99,18 @@ erDiagram
...
+ address client_handle
+ address node_handle
+ address rmw_client_handle
+ string service_name
+ }
+
+ rcl_lifecycle_state_machine_init{
+ address node_handle
+ address state_machine
  }
```
[the remark this answers is not in the document]
Comment (xygyo77) [reply docs/design/trace_points/initialization_trace_points.md:114]: Yes, I do.

File: docs/design/event_and_latency_definitions/publisher.md (hunk 2 above)
Comment (xygyo77) [reply docs/design/event_and_latency_definitions/publisher.md:83]: I thought that the dds_bind_addr_to_stamp trace point was not shown in cyclonedds and humble until you add it to rcl_publish_serialised_message() even if you do generic_publish(). I thought that would be added after jazzy, but am I wrong?
For fastdds, the dds_bind_addr_to_stamp trace point does appear, right?
Separately, did the dds_bind_addr_to_addr trace point actually output?

File: docs/design/trace_points/runtime_trace_points.md
Comment (xygyo77) [reply docs/design/trace_points/runtime_trace_points.md:1]: I agree.

File: docs/design/event_and_latency_definitions/publisher.md (hunk 2 above)
Comment (xygyo77) [reply docs/design/event_and_latency_definitions/publisher.md:83]: In generic_publish(), dds_bind_addr_to_stamp should be output, so this ‘Info’ is dropped.

**Comment by ymski** [comment]
We should also add `rmw_take` to the list of trace events for `docs/design/event_and_latency_definitions/communication.md` L145, `docs/design/event_and_latency_definitions/subscription.md` L93.
Like dispatch_subscription_callback, rmw_take is used to link published data with subscribed data using source_timestamp.

File: docs/design/trace_points/runtime_trace_points.md
[the remark this answers is not in the document]
Comment (xygyo77) [reply docs/design/trace_points/runtime_trace_points.md:1]: Yes, to be added.

**Comment by ymski** [comment]
> We should also add `rmw_take` to the list of trace events for `docs/design/event_and_latency_definitions/communication.md` L145, `docs/design/event_and_latency_definitions/subscription.md` L93. Like dispatch_subscription_callback, rmw_take is used to link published data with subscribed data using source_timestamp.

@xygyo77 
Sorry. My comment was incorrect.
`rmw_take` is used for inter process communication.
So L55 is the place to modify, not L93 in subscription.md.

**Comment by ymski** [comment]
This is a good opportunity to organize the description of intra-process communication from iron onward.
Please add trace points to the list in the file below.

- docs/design/event_and_latency_definitions/communication.md L145
  - rclcpp_ring_buffer_enqueue
  - rclcpp_ring_buffer_dequeue
- docs/design/event_and_latency_definitions/publisher.md L71
  - rclcpp_ring_buffer_enqueue
- docs/design/event_and_latency_definitions/subscription.md L93
  - rclcpp_ring_buffer_dequeue

And, in several places dds_bind_addr_to_stamp seems to be abbreviated as bind_addr_to_stamp.
This is also a good time to fix it.
Two such tracing points that I am aware of are as follows.
- dds_bind_addr_to_stamp (abbreviated as `bind_addr_to_stamp`)
- dds_bind_addr_to_addr (abbreviated as `bind_addr_to_addr`)

**Review by ymski — APPROVED** [review]
LGTM!

**Pull request merged**